src/caffe/layers/accuracy_layer.cpp

```diff
@@ -82,6 +95,18 @@ void AccuracyLayer<Dtype>::Forward_cpu(const vector<Blob<Dtype>*>& bottom,
 
   // LOG(INFO) << "Accuracy: " << accuracy;
   top[0]->mutable_cpu_data()[0] = accuracy / count;
+  if (top.size() > 1) {
+    int output_label = 0;
+    for (int i = 0; i < dim; ++i) {
+      if (has_ignore_label_ && i == ignore_label_) {
+        continue;
+      }
+      top[1]->mutable_cpu_data()[output_label] =
+          nums_buffer_.cpu_data()[i] == 0 ? 0
+          : top[1]->cpu_data()[i] / nums_buffer_.cpu_data()[i];
+      ++output_label;
+    }
+  }
```

Comment:
I see what you mean here, but it seems confusing to shift label index in `top[1]` due to ignore_label.

Better to keep original index in `top[1]` and simply do

    if (top.size() > 1) {
      int top_count = top[1]->count();
      for (int i = 0; i < top_count; ++i) {
        top[1]->mutable_cpu_data()[i] =
            nums_buffer_.cpu_data()[i] == 0 ? 0
            : top[1]->cpu_data()[i] / nums_buffer_.cpu_data()[i])
      }
    }